opal/mca/btl/tcp/btl_tcp.c

```diff
@@ -330,151 +329,6 @@ int mca_btl_tcp_send(struct mca_btl_base_module_t *btl, struct mca_btl_base_endp
     return mca_btl_tcp_endpoint_send(endpoint, frag);
 }
 
-static void fake_rdma_complete(mca_btl_base_module_t *btl, mca_btl_base_endpoint_t *endpoint,
-                               mca_btl_base_descriptor_t *desc, int rc)
-{
-    mca_btl_tcp_frag_t *frag = (mca_btl_tcp_frag_t *) desc;
-
-    frag->cb.func(btl, endpoint, frag->segments[0].seg_addr.pval, NULL, frag->cb.context,
-                  frag->cb.data, rc);
-}
-
-/**
- * Initiate an asynchronous put.
- */
-
-int mca_btl_tcp_put(mca_btl_base_module_t *btl, struct mca_btl_base_endpoint_t *endpoint,
```

Comment:
Without this how is the TCP BTL providing support for PUT ?

Reply:
via active messages in btl/base as implemented here:

https://github.com/open-mpi/ompi/pull/8493/commits/3e317753f0308b917b4aa3b93b2d92997b9f491f

basically it'll now go down the btl/base path. TCP GET was already going down this path as far as I could see, as it was unused in btl/tcp.

The current btl/tcp is incorrect in that it will give wring answers, I provided an alternate solution in #8700. Both work for purposes of fixing the bad behavior.